scripts/record_bag_pnc.sh

```diff
@@ -0,0 +1,106 @@
+#!/usr/bin/env bash
+
+###############################################################################
+# Copyright 2017 The Apollo Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+###############################################################################
+
+
+DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
+
+source "${DIR}/apollo_base.sh"
+
+function start() {
+  BAG_DIR="${APOLLO_ROOT_DIR}/data/bag"
+
+  # Record bag to the largest portable-disk.
+  if [ "$1" = "--portable-disk" ]; then
+    LARGEST_DISK="$(df | grep "/media/${DOCKER_USER}" | sort -nr -k 4 | \
+        awk '{print substr($0, index($0, $6))}')"
+    if [ ! -z "${LARGEST_DISK}" ]; then
+      REAL_BAG_DIR="${LARGEST_DISK}/data/bag"
+      if [ ! -d "${REAL_BAG_DIR}" ]; then
+        mkdir -p "${REAL_BAG_DIR}"
+      fi
+      BAG_DIR="${APOLLO_ROOT_DIR}/data/bag/portable"
+      rm -fr "${BAG_DIR}"
+      ln -s "${REAL_BAG_DIR}" "${BAG_DIR}"
+    else
+      echo "Cannot find portable disk."
+      echo "Please make sure your container was started AFTER inserting the disk."
+    fi
+  fi
+
+  # Create and enter into bag dir.
+  if [ ! -e "${BAG_DIR}" ]; then
+    mkdir -p "${BAG_DIR}"
+  fi
+  cd "${BAG_DIR}"
+  echo "Recording bag to: $(pwd)"
+
+  # Start recording.
+  LOG="/tmp/apollo_record.out"
+  NUM_PROCESSES="$(pgrep -c -f "rosbag record")"
+  if [ "${NUM_PROCESSES}" -eq 0 ]; then
+    nohup rosbag record -b 2048  \
+        /apollo/sensor/gnss/gnss_status \
+        /apollo/sensor/gnss/odometry \
+        /apollo/sensor/gnss/ins_stat \
+        /apollo/sensor/gnss/corrected_imu \
+        /apollo/sensor/mobileye \
+        /apollo/sensor/delphi_esr \
```

Comment:
I refer to this line, do we still record delphi_esr by default?

Reply:
yes.